scheduler/pkg/server/server_status.go

```diff
@@ -173,6 +173,24 @@ func (s *SchedulerServer) handleModelEventForServerStatus(event coordinator.Mode
 	}
 }
 
+func (s *SchedulerServer) handleServerEvents(event coordinator.ServerEventMsg) {
+	logger := s.logger.WithField("func", "handleServerEvent")
+	logger.Debugf("Got server state %s change for %s", event.ServerName, event.String())
+
+	server, err := s.modelStore.GetServer(event.ServerName, true, true)
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to get server %s", event.ServerName)
+		return
+	}
+
+	if event.UpdateContext == coordinator.SERVER_SCALE {
```

Comment:
@driev this is the current proposed location to consume server scale (up / down) events and decide on whether actually to change server replicas. Open to suggestions if there are better places for this logic to be defined.

Reply:
For example `SchedulerServer` could be extended with a `ServerScaler` utility where the logic of scaling up and down could be defined. This is perhaps something we can consider as a follow up PR after we get scale up / down done?

Reply:
Yeah, since it's likely to change again, it's probably better to wait a bit.